### ISSUE TITLE
Distinguish meta CLI commands like `version` and `completion` in docs site

### ISSUE THREAD

**Issue by david-crespo** [oxidecomputer/oxide.rs#594]
### Target component

- [X] CLI
- [ ] SDK
- [ ] Something else
- [ ] Not sure

### Overview

It would be nice to put information in the CLI docs JSON to allow the docs site to distinguish between commands like `completion` and `version` (possibly `auth`? meh, nah) and the rest, which have to do with actual API resources.

<img width="506" alt="image" src="https://github.com/oxidecomputer/oxide.rs/assets/3612203/25c5e17f-a7be-4d24-a916-d001d6337f73">

### Implementation details

Two ideas:

* Give tags to top-level CLI commands to allow grouping in docs site sidebar, like we do with the API endpoints. In the API, endpoints with a tag starting with `system/` get grouped under `System` and the rest go under `Developer`.
* Don't do formal grouping or tagging, but put `completions` and `version` under a top-level `oxide self` command so they at least take up less space and it's easier to tell they don't have anything to do with the API itself.

**Comment by karencfv** [comment]
> Don't do formal grouping or tagging, but put completions and version under a top-level oxide self command so they at least take up less space and it's easier to tell they don't have anything to do with the API itself.

I'm a little hesitant to remove `completion` and `version` from the top level commands. These are (by convention?) generally top level commands or global flags in most CLIs (I'd like to say all but I haven't used all CLIs that exist), so it's not unreasonable to say that users will likely be expecting these commands to be top level commands or global flags.

**Comment by ahl** [comment]
I like them as top level commands... and I think there might be opportunities to separate them out both in the generated docs as well as in the help message. Good idea, @david-crespo.

**Comment by david-crespo** [comment]
Yeah, I'm persuaded that keeping them top level and doing grouping is better.
